assets/js/modules/analytics-4/components/widgets/TopTrafficSourceWidget.js

```diff
@@ -21,27 +21,123 @@
  */
 import PropTypes from 'prop-types';
 
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+
 /**
  * Internal dependencies
  */
 import Data from 'googlesitekit-data';
+import MetricTileText from '../../../../components/KeyMetrics/MetricTileText';
 import { CORE_USER } from '../../../../googlesitekit/datastore/user/constants';
+import {
+	DATE_RANGE_OFFSET,
+	MODULES_ANALYTICS_4,
+} from '../../datastore/constants';
 
-const { useSelect } = Data;
+const { useSelect, useInViewSelect } = Data;
 
 export default function TopTrafficSourceWidget( { Widget, WidgetNull } ) {
 	const keyMetricsWidgetHidden = useSelect( ( select ) =>
 		select( CORE_USER ).isKeyMetricsWidgetHidden()
 	);
 
+	const dates = useSelect( ( select ) =>
+		select( CORE_USER ).getDateRangeDates( {
+			offsetDays: DATE_RANGE_OFFSET,
+			compare: true,
+		} )
+	);
+
+	const totalUsersReportOptions = {
+		...dates,
+		metrics: [
+			{
+				name: 'totalUsers',
+			},
+		],
+	};
+
+	const trafficSourceReportOptions = {
+		...dates,
+		dimensions: [ 'sessionDefaultChannelGroup' ],
+		metrics: [
+			{
+				name: 'totalUsers',
+			},
+		],
+		limit: 1,
+		orderBy: 'totalUsers',
+	};
+
+	const totalUsersReport = useInViewSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getReport( totalUsersReportOptions )
+	);
+
+	const trafficSourceReport = useInViewSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getReport( trafficSourceReportOptions )
+	);
+
+	const loading = useSelect(
+		( select ) =>
+			! select( MODULES_ANALYTICS_4 ).hasFinishedResolution(
+				'getReport',
+				[ totalUsersReportOptions ]
+			) ||
+			! select( MODULES_ANALYTICS_4 ).hasFinishedResolution(
+				'getReport',
+				[ trafficSourceReportOptions ]
+			)
+	);
+
+	const totalUsers =
+		parseInt(
+			totalUsersReport?.rows?.[ 1 ]?.metricValues[ 0 ]?.value,
+			10
+		) || 0;
+
+	const topTrafficSource =
+		trafficSourceReport?.rows?.[ 1 ]?.dimensionValues[ 0 ]?.value || '-';
+
+	const currentTopTrafficSourceUsers =
+		parseInt(
+			trafficSourceReport?.rows?.[ 1 ]?.metricValues[ 0 ]?.value,
+			10
+		) || 0;
+
+	const topTrafficSourceUsersPercentage = totalUsers
+		? ( currentTopTrafficSourceUsers * 100 ) / totalUsers
```

Comment:
No need to multiply by 100 here, `numFmt` will property format this value.
```suggestion
		? currentTopTrafficSourceUsers / totalUsers
```